smi/adaptor/controller.go

```diff
@@ -0,0 +1,340 @@
+			// As the item in the workqueue is actually invalid, we call
+			// Forget here else we'd go into a loop of attempting to
+			// process a work item that is invalid.
+			c.workqueue.Forget(obj)
+			log.Errorf("expected string in workqueue but got %#v", obj)
+			return nil
+		}
+		// Run the syncHandler, passing it the namespace/name string of the
+		// Ts resource to be synced.
+		if err := c.syncHandler(key); err != nil {
+			// Put the item back on the workqueue to handle any transient errors.
+			c.workqueue.AddRateLimited(key)
+			return fmt.Errorf("error syncing '%s': %s, requeuing", key, err.Error())
+		}
+		// Finally, if no error occurs we Forget this item so it does not
+		// get queued again until another change happens.
+		c.workqueue.Forget(obj)
+		log.Infof("Successfully synced '%s'", key)
+		return nil
+	}(obj)
+
+	if err != nil {
+		log.Error(err)
+		return true
+	}
+
+	return true
+}
+
+// syncHandler compares the actual state with the desired, and attempts to
+// converge the two. It then updates the Status block of the Ts resource
+// with the current status of the resource.
+func (c *SMIController) syncHandler(key string) error {
+	// Convert the namespace/name string into a distinct namespace and name
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		log.Errorf("invalid resource key: %s", key)
+		return nil
+	}
+
+	// Get the Ts resource with this namespace/name
+	ts, err := c.tsLister.TrafficSplits(namespace).Get(name)
+	if err != nil {
+		// Check if TS does not exit anymore
+		if errors.IsNotFound(err) {
+			// Check if there is a relevant SP that was created or updated by SMI Controller
+			selector := labels.NewSelector()
+			r, err := labels.NewRequirement(createdOrUpdatedBySMIAdaptorFor, selection.Equals, []string{name})
+			if err != nil {
+				return err
+			}
+			selector.Add(*r)
+			sps, err := c.spLister.ServiceProfiles(namespace).List(selector)
+			if err != nil {
+				return err
+			}
+
+			if len(sps) != 0 {
+				// Empty dstOverrides in the SP
+				sp := sps[0]
+				if ignoreAnnotationPresent(sp) {
+					log.Infof("skipping %s sp as ignore annotation is present", sp.Name)
+					return nil
+				}
+
+				sp.Spec.DstOverrides = nil
+				_, err = c.spclientset.LinkerdV1alpha2().ServiceProfiles(namespace).Update(context.Background(), sp, metav1.UpdateOptions{})
+				if err != nil {
+					return err
+				}
+			}
+			return nil
+		}
+		return err
+	}
+
+	// Check if the Service Profile is already Present
+	sp, err := c.spLister.ServiceProfiles(ts.Namespace).Get(fmt.Sprintf("%s.%s.svc.%s", ts.Spec.Service, ts.Namespace, c.clusterDomain))
+	if err != nil {
+		// Create a Service Profile resource as it does not exist
+		_, err = c.spclientset.LinkerdV1alpha2().ServiceProfiles(ts.Namespace).Create(context.Background(), c.toServiceProfile(ts), metav1.CreateOptions{})
+		if err != nil {
+			return err
+		}
+	} else {
+		log.Infof("Service Profile already present")
```

Comment:
I recommend adding the sp name here:
```suggestion
		log.Infof("Service Profile [%s] already present", sp.Name)
```

Also, at `INFO` level, the logs would just have this message repeated, maybe that's ok?

Reply:
Yep, This seems like a useful log to have! :thinking: